types/index.d.ts

```diff
@@ -4,6 +4,15 @@ import {Attributes, Element, Node} from 'xast'
 
 type Children = string | Node | Children[]
 
+type Primitive = null | undefined | string | number | boolean | symbol | bigint;
```

Comment:
@wooorm to confirm, does that API intent do support all primitives? Or some subset of primitives?

Reply:
@malav2110 I like where this is going but I think it’s bit more verbose than it needs to be in my opinion.

xastscript accepts any value as attribute values. But, xast expects strings, so `xastscript` will cast everything for you to a string (it’s a bit useful, right, that in JS you don‘t *have* to cast a date to a string yourself, but xastscript does the expected thing for you?). It’s not really suggested, but adding exceptions for symbols, bigints, etc, is also a bit too much for me: that’s what TS can help you with if you want it to.

Now, `null | undefined` are specifically supported: they are ignored. `number` is something that is explicitly handled here too (to detect if something is a child instead of attributes, but also to create a text node from a child). But I feel adding `boolean | symbol | bigint` here is a bit too much? If you’re in TS, cast your values to string on your side, and everything will work perfectly!

```suggestion
type Primitive = null | undefined | string | number;
```

Reply:
@wooorm Updating the PR to accomodate your suggestions